src/auro-calendar.js

```diff
@@ -0,0 +1,339 @@
+import { html } from 'lit';
+import styleCss from "./style-auro-calendar-css";
+
+import './auro-calendar-month.js';
+// import { RangeDatepicker } from '../node_modules/wc-range-datepicker/dist/src/range-datepicker';
+import { RangeDatepicker } from './../vendor/wc-range-datepicker/range-datepicker';
+
+// See https://git.io/JJ6SJ for "How to document your components using JSDoc"
+/**
+ * @prop {Object} firstDayOfWeek - Weekday that will be displayed in first column of month grid.
+ * 0: sunday, 1: monday, 2: tuesday, 3: wednesday , 4: thursday, 5: friday, 6: saturday
+ * Default is 0.
+ * @prop {Date | null} focusedDate - The currently focused date (if any).
+ * @prop {Date} maxDate - Maximum date. All dates after will be disabled.
+ * @prop {Date} minDate - Minimum date. All dates before will be disabled.
+ * @prop {Date | undefined} selectedDate - The selected date, usually synchronized with datepicker-input.
+ * Not to be confused with the focused date (therefore not necessarily in active month view).
+ * @prop {string} weekdayHeaderNotation - Weekday header notation, based on Intl DatetimeFormat:
+ * - 'short' (e.g., Thu)
+ * - 'narrow' (e.g., T).
+ * Default is 'short'.
+ * @fires auroCalendar-dateSelected - Notifies that a date has been selected in the calendar.
+ * @fires auroCalendar-ready - Notifies that the component has finished initializing.
+ */
+
+/* eslint-disable no-self-assign, no-magic-numbers, no-undef-init, no-param-reassign, max-lines */
+
+// class AuroCalendar extends LitElement {
+export class AuroCalendar extends RangeDatepicker {
+  constructor() {
+    super();
+
+    this.numCalendars = 1;
+    this.showPrevMonthBtn = true;
+    this.showNextMonthBtn = true;
+  }
+
+  static get styles() {
+    return [
+      // ...super.styles,
+      styleCss
+    ];
+  }
+
+  static get properties() {
+    return {
+      // ...super.properties,
+      numCalendars: {
+        type: Number
+      },
+      dateFrom: {
+        type: String
+      },
+      dateTo: {
+        type: String
+      },
+      maxDate: {
+        type: String,
+        reflect: true
+      },
+      minDate: {
+        type: String,
+        reflect: true
+      }
+    };
+  }
+
+  requestDismiss() {
+    this.dispatchEvent(new CustomEvent('auroCalendar-dismissRequest', {
+      bubbles: true,
+      cancelable: false,
+      composed: true,
+    }));
+  }
+
+  /**
+   * @private
+   * @returns {void} Marks the component as ready and sends event.
+   */
+  notifyReady() {
+    this.ready = true;
+
+    this.dispatchEvent(new CustomEvent('auroCalendar-ready', {
+      bubbles: true,
+      cancelable: false,
+      composed: true,
+    }));
+  }
```

Comment:
This and `renderCalendar` are the only code blocks that are properly documented.